Add docker-compose create command.

seguins opened the pull request:
Adds a new command  for creating a service without starting it.

Closes #1125

aanand commented:
LGTM

dnephin commented on compose/project.py:
```diff
@@ -215,6 +221,22 @@ def build(self, service_names=None, no_cache=False, pull=False, force_rm=False):
+                service.create_container(do_build=True)
+            elif plan.action == 'recreate':
+                service.recreate_container(
+                    service.containers(stopped=True)[0],
```
I don't think this is right, there could be more than one container at some point.

seguins replied:
Indeed, it will be removed and fixed by the call to ```service.execute_convergence_plan```

dnephin commented on compose/project.py:
```diff
@@ -215,6 +221,22 @@ def build(self, service_names=None, no_cache=False, pull=False, force_rm=False):
+                    service.containers(stopped=True)[0],
+                    do_build=do_build,
+                    start_new_container=False
+                )
```
I think it would be good to keep this logic in `service.py`, since it's very similar to the existing `execute_convergence_plan`. Maybe `service.execute_create_convergence_plan()` ? It's a bit wordy, but at least it's clear.

aanand replied:
`service. execute_convergence_plan(start=False)`?

seguins replied:
@dnephin Thanks for the feedback, I will apply the solution of @aanand which seems cleaner.

dnephin replied:
I'm a little concerned that is going to make `execute_convergence_plan()` a little overly complex.  We have a max-complexity set to 12, so I guess as long as we don't have to relax that I'm cool with it.

dnephin commented on tests/acceptance/cli_test.py:
```diff
@@ -264,6 +264,74 @@ def test_build_failed_forcerm(self):
+        self.assertEqual(len(db.containers()), 0)
+        self.assertEqual(len(web.containers(stopped=True)), 1)
+        self.assertEqual(len(db.containers(stopped=True)), 1)
+        self.assertEqual(len(console.containers(stopped=True)), 0)
```
Following the [testing pyramid](http://martinfowler.com/bliki/TestPyramid.html) I think the above two tests (create_twice and with_link) should be integration tests in `tests/integration/`. We already have plenty of end-to-end coverage with the other tests you've added.

seguins commented:
@dnephin thanks for the review.
I updated the commit : 
* ```project.create``` uses now ```service.execute_convergence_plan```
* I removed two acceptance tests and add new integration tests

dnephin commented on compose/project.py:
```diff
@@ -215,6 +221,14 @@ def build(self, service_names=None, no_cache=False, pull=False, force_rm=False):
+        plans = self._get_convergence_plans(services, strategy)
+
+        for service in services:
+            service.execute_convergence_plan(plans[service.name], do_build, start=False)
```
Just one more small thing, I think we want to avoid calling attach here (it's an unnecessary API call), so this should add `detached=True`  as well

dnephin commented:
Looks good, just one more small thing

seguins commented:
Done

dnephin commented:
LGTM

thaJeztah commented:
@seguins looks like this needs a rebase :cry:

thaJeztah commented:
Oh, maybe not, it doesn't say "merge conflict", it's probably github being over restrictive (sorry for the noise :smile:)
